extensions/wikia/VideoHandlers/maintenance/uploadVideo.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * This script will upload Video to a wiki.
+ */
+
+require_once __DIR__ . '/../../../../maintenance/Maintenance.php';
+
+/**
+ * Maintenance script class
+ */
+class UploadVideo extends Maintenance {
+
+	/**
+	 * Set script options
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->mDescription = 'This script uploads video to a wiki';
+
+		$this->addOption( 'file-list', "List of files to upload in format 'provide:filename'", true );
+	}
+
+	/**
+	 * @param string $fileName
+	 * @return mixed
+	 */
+	private function uploadVideo( $fileName ) {
+		$videoService = new VideoService();
+		return $videoService->addVideo( $fileName );
+	}
+
+	public function execute() {
+		global $wgUser, $wgDBname;
+
+		$fileList = realpath( $this->getOption( 'file-list' ) );
+		if ( !file_exists( $fileList ) ) {
+			$this->error( "File with list of files to upload does not exists: $fileList", 1 );
+		}
+
+		$files = file( $fileList );
```

Comment:
Minor: I'd consider passing `FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES` as the second param here. You're trimming so `FILE_IGNORE_NEW_LINES` isn't strictly needed, but `FILE_SKIP_EMPTY_LINES` will avoid us trying to upload an empty string if we have a new line at the end of the file.

Reply:
Fix in: 13f43a51314